File: Windows Phone 8.1/2.5.7.0/Decompilation/Vine.WindowsPhone/Vine.Models.Analytics/DeviceData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Newtonsoft.Json;
using NodaTime;
using Windows.Devices.Sensors;
using Windows.Networking.Connectivity;
using Windows.Phone.Devices.Power;
using Windows.Security.ExchangeActiveSyncProvisioning;
using Windows.System.UserProfile;

namespace Vine.Models.Analytics;

public class DeviceData
{
	[JsonProperty("os")]
	public string OS { get; set; }

	[JsonProperty("os_version")]
	public string OSVersion { get; set; }

	[JsonProperty("manufacturer")]
	public string Manufacturer { get; set; }

	[JsonProperty("device_name")]
	public string DeviceName { get; set; }

	[JsonProperty("device_model")]
	public string DeviceModel { get; set; }

	[JsonProperty("battery_level")]
	public double BatteryLevel { get; set; }

	[JsonProperty("brightness")]
	public double? Brightness { get; set; }

	[JsonProperty("orientation")]
	public ScribeOrientation? Orientation { get; set; }

	[JsonProperty("internet_access_type")]
	public InternetAccessType InternetAccessType { get; set; }

	[JsonProperty("language_codes")]
	public string[] LanguageCodes { get; set; }

	[JsonProperty("radio_details")]
	public MobileRadioDetails RadioDetails { get; set; }

	[JsonProperty("timezone")]
	public string TimeZone { get; set; }

	public static DeviceData GetDefault()
	{
		DeviceData deviceData = new DeviceData();
		try
		{
			SetDeviceInfo(deviceData);
			deviceData.OSVersion = ApplicationSettings.Current.OSVersion;
			deviceData.TimeZone = GetTimeZone();
			deviceData.InternetAccessType = GetInternetAccessType();
			deviceData.LanguageCodes = GlobalizationPreferences.Languages.ToArray();
			string mCC = ApplicationSettings.Current.MCC;
			string mNC = ApplicationSettings.Current.MNC;
			if (deviceData.InternetAccessType == InternetAccessType.MOBILE && !string.IsNullOrEmpty(mCC))
			{
				deviceData.RadioDetails = new MobileRadioDetails
				{
					MobileNetworkOperatorCode = mNC,
					MobileNetworkOperatorIsoCountryCode = mCC,
					MobileNetworkOperatorName = GetMobileNetworkOperatorName()
				};
			}
		}
		catch (Exception)
		{
		}
		return deviceData;
	}

	private static string GetTimeZone()
	{
		try
		{
			return DateTimeZoneProviders.Tzdb.GetSystemDefault().Id;
		}
		catch (Exception)
		{
			return null;
		}
	}

	private static void SetDeviceInfo(DeviceData deviceData)
	{
		//IL_0000: Unknown result type (might be due to invalid IL or missing references)
		//IL_0006: Expected O, but got Unknown
		try
		{
			EasClientDeviceInformation val = new EasClientDeviceInformation();
			deviceData.Manufacturer = val.SystemManufacturer;
			deviceData.DeviceName = val.SystemSku;
			deviceData.DeviceModel = val.SystemProductName;
			deviceData.OS = val.OperatingSystem;
			deviceData.BatteryLevel = (float)Battery.GetDefault().RemainingChargePercent / 100f;
		}
		catch (Exception)
		{
		}
	}

	private static ScribeOrientation GetDeviceOrientation()
	{
		//IL_000a: Unknown result type (might be due to invalid IL or missing references)
		//IL_000f: Unknown result type (might be due to invalid IL or missing references)
		//IL_0010: Unknown result type (might be due to invalid IL or missing references)
		//IL_002a: Expected I4, but got Unknown
		try
		{
			SimpleOrientationSensor val = SimpleOrientationSensor.GetDefault();
			if (val != null)
			{
				SimpleOrientation currentOrientation = val.GetCurrentOrientation();
				switch ((int)currentOrientation)
				{
				case 0:
				case 2:
					return ScribeOrientation.PORTRAIT;
				case 1:
				case 3:
					return ScribeOrientation.LANDSCAPE;
				case 4:
					return ScribeOrientation.FACE_UP;
				}
			}
		}
		catch (Exception)
		{
		}
		return ScribeOrientation.PORTRAIT;
	}

	private static InternetAccessType GetInternetAccessType()
	{
		try
		{
			ConnectionProfile internetConnectionProfile = NetworkInformation.GetInternetConnectionProfile();
			if (internetConnectionProfile != null)
			{
				if (internetConnectionProfile.IsWlanConnectionProfile)
				{
					return InternetAccessType.WIFI;
				}
				if (internetConnectionProfile.IsWwanConnectionProfile)
				{
					return InternetAccessType.MOBILE;
				}
			}
		}
		catch (Exception)
		{
		}
		return InternetAccessType.UNREACHABLE;
	}

	private static string GetMobileNetworkOperatorName()
	{
		try
		{
			ConnectionProfile internetConnectionProfile = NetworkInformation.GetInternetConnectionProfile();
			if (internetConnectionProfile != null && internetConnectionProfile.IsWwanConnectionProfile)
			{
				using IEnumerator<string> enumerator = internetConnectionProfile.GetNetworkNames().GetEnumerator();
				if (enumerator.MoveNext())
				{
					return enumerator.Current;
				}
			}
		}
		catch (Exception)
		{
		}
		return null;
	}
}

File: Android/Vine-5.50.1/Decompilation/app/src/main/java/co/vine/android/scribe/model/DeviceData.java
package co.vine.android.scribe.model;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;
import java.util.List;

@JsonObject
/* loaded from: classes.dex */
public class DeviceData {

    @JsonField(name = {"battery_level"})
    public Double batteryLevel;

    @JsonField(name = {"brightness"})
    public Double brightness;

    @JsonField(name = {"browser"})
    public String browser;

    @JsonField(name = {"browser_version"})
    public String browserVersion;

    @JsonField(name = {"bytes_available"})
    public Long bytesAvailable;

    @JsonField(name = {"bytes_free"})
    public Long bytesFree;

    @JsonField(name = {"device_model"})
    public String deviceModel;

    @JsonField(name = {"device_name"})
    public String deviceName;

    @JsonField(name = {"location"})
    public GPSData gpsData;

    @JsonField(name = {"internet_access_type"})
    public String internetAccessType;

    @JsonField(name = {"language_codes"})
    public List<String> languageCodes;

    @JsonField(name = {"manufacturer"})
    public String manufacturer;

    @JsonField(name = {"orientation"})
    public String orientation;

    @JsonField(name = {"os"})
    public String os;

    @JsonField(name = {"os_version"})
    public String osVersion;

    @JsonField(name = {"other_audio_is_playing"})
    public Boolean otherAudioIsPlaying;

    @JsonField(name = {"radio_details"})
    public MobileRadioDetails radioDetails;

    @JsonField(name = {"timezone"})
    public String timezone;

    public enum InternetAccessType {
        UNREACHABLE,
        MOBILE,
        WIFI,
        HARDLINE,
        OTHER
    }

    public enum Orientation {
        PORTRAIT,
        LANDSCAPE,
        FACE_UP
    }
}
